zigpy/device.py

```diff
@@ -306,7 +328,28 @@ def manufacturer_id(self) -> Optional[int]:
 
     @property
     def model(self):
-        return self._model
+        if self._model is not None:
+            return self._model
+
+        for endpoint_id, endpoint in self.endpoints.items():
+            if endpoint_id == 0:
+                continue
+
+            if Basic.cluster_id not in endpoint.in_clusters:
+                continue
+
+            cluster = endpoint.in_clusters[Basic.cluster_id]
+
+            if 0x0005 not in cluster._attr_cache:
+                continue
+
+            value = cluster._attr_cache[0x0005]
+
+            if isinstance(value, bytes):
+                value = value.split(b"\x00")[0]
+                return value.decode().strip()
+            else:
+                return value
```

Comment:
same. as we're not expecting this to change

Reply:
I've added a `Device._update_model_manufacturer` method that is called when a new device is initialized or loaded from the DB.

Reply:
maybe just make this a private method, pass it the attr you are looking for and then set the `self._manufacturer` or `self.model` to the returned value.
So there's no code duplication and we don't run this every time the property is checked

Reply:
Like `_update_model_manufacturer` is now doing (https://github.com/zigpy/zigpy/pull/521/files#diff-62448beb11fb75efec436d695b9dda880fb688b152a4969aa32e501124072072R393-R402)

It'll be run only once (at most twice I think?) per device. You can see this in the unit test: https://github.com/zigpy/zigpy/pull/521/files#diff-72107d6f4a703abaeb6a7cf709e0c99f7ed0b7c74f0b14b0cbc4f35c7c650e26R309-R337